spring/boot2-webflux-autoconfigure/src/main/java/com/linecorp/armeria/spring/web/reactive/ArmeriaReactiveWebServerFactory.java

```diff
@@ -134,84 +133,118 @@ public WebServer getWebServer(HttpHandler httpHandler) {
             return armeriaWebServerBean;
         }
 
+        final Http2 http2 = getHttp2();
+        if (http2 != null && !http2.isEnabled()) {
+            logger.warn("Cannot disable HTTP/2 protocol for Armeria server. It will be enabled automatically.");
+        }
+
         final ServerBuilder sb = Server.builder();
         sb.disableServerHeader();
         sb.disableDateHeader();
 
-        final SessionProtocol protocol;
-        final Ssl ssl = getSsl();
-        if (ssl != null) {
-            if (ssl.isEnabled()) {
-                final SslStoreProvider provider = getSslStoreProvider();
-                final Supplier<KeyStore> keyStoreSupplier;
-                final Supplier<KeyStore> trustStoreSupplier;
-                if (provider != null) {
-                    keyStoreSupplier = () -> {
-                        try {
-                            return provider.getKeyStore();
-                        } catch (Exception e) {
-                            throw new IllegalStateException(e);
-                        }
-                    };
-                    trustStoreSupplier = () -> {
-                        try {
-                            return provider.getTrustStore();
-                        } catch (Exception e) {
-                            throw new IllegalStateException(e);
-                        }
-                    };
-                } else {
-                    keyStoreSupplier = null;
-                    trustStoreSupplier = null;
-                }
-                configureTls(sb, toArmeriaSslConfiguration(ssl), keyStoreSupplier, trustStoreSupplier);
-                protocol = SessionProtocol.HTTPS;
-            } else {
-                logger.warn("TLS configuration exists but it is disabled by 'enabled' property.");
-                protocol = SessionProtocol.HTTP;
+        final ArmeriaSettings armeriaSettings = findBean(ArmeriaSettings.class);
+        if (!isArmeriaCompressionEnabled(armeriaSettings)) {
+            final Compression compression = getCompression();
+            if (compression != null && compression.getEnabled()) {
+                final long minResponseSize = compression.getMinResponseSize().toBytes();
+                sb.decorator(contentEncodingDecorator(compression.getMimeTypes(),
+                                                      compression.getExcludedUserAgents(),
+                                                      Ints.saturatedCast(minResponseSize)));
```

Comment:
How about just converting Spring `Compression` to Armeria `Compression` and set it to `ArmeriaSettings` without creating the encoding decorator in order to centralize the auto configuration logic?

Reply:
Thanks for the suggestion. Let me do it when I refactor `AbstractArmeriaAutoConfiguration` in the next PR. 😄